src/applications/vaos/tests/project-cheetah/components/NewBookingPage.unit.spec.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { expect } from 'chai';
+import { mockFetch, resetFetch } from 'platform/testing/unit/helpers';
+import { createTestStore, renderWithStoreAndRouter } from '../../mocks/setup';
+import { NewBooking } from '../../../project-cheetah';
+import { getDirectBookingEligibilityCriteriaMock } from '../../../tests/mocks/v0';
+import { mockDirectBookingEligibilityCriteria } from '../../../tests/mocks/helpers';
+
+const initialState = {
+  featureToggles: {
+    vaOnlineSchedulingDirect: true,
+  },
+  user: {
+    profile: {
+      facilities: [
+        { facilityId: '983', isCerner: false },
+        { facilityId: '984', isCerner: false },
+      ],
+    },
+  },
+};
+
+describe('VAOS <ConfirmedAppointmentDetailsPage>', () => {
+  beforeEach(() => {
+    mockFetch();
+  });
+
+  afterEach(() => {
+    resetFetch();
+  });
+
+  it('should not redirect the user to the Contact Facilities page', async () => {
+    const store = createTestStore({
+      ...initialState,
+    });
+
+    const response = getDirectBookingEligibilityCriteriaMock();
+    response.attributes.coreSettings[0].id = '301';
+    response.attributes.coreSettings[0].typeOfCare = 'Vaccine';
+
+    mockDirectBookingEligibilityCriteria(
+      ['983', '984'],
+      [
+        {
+          ...response,
+        },
+      ],
+    );
+
+    const screen = renderWithStoreAndRouter(<NewBooking />, {
+      store,
+    });
+
+    await screen.findByRole('heading', { level: 1, name: 'Plan ahead' });
+
+    expect(screen.history.push.lastCall.args[0]).to.equal('/');
+  });
+
+  it('should redirect the user to the Contact Facilities page', async () => {
+    const store = createTestStore({
+      ...initialState,
+    });
+
+    const response = getDirectBookingEligibilityCriteriaMock();
+    delete response.attributes.coreSettings[0].patientHistoryRequired;
+
+    mockDirectBookingEligibilityCriteria(
+      ['983', '984'],
+      [
+        {
+          ...response,
+        },
+      ],
+    );
+
+    const screen = renderWithStoreAndRouter(<NewBooking />, {
+      store,
+    });
+
+    await screen.findByRole('heading', { level: 1, name: 'Plan ahead' });
```

Comment:
Follow up to in person walkthrough: I think you can change this to look for the heading on the contact facilities page.

Reply:
This change doesn't work since then page change never happens. I believe this is due to history.push being a mock (setup.js:62). So, I modified to this: 

```
// Wait for the redirect.

await waitFor(() => expect(screen.history.push.lastCall.args[0]).to.equal('/new-project-cheetah-booking/contact-facilities',),);
```